src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs

```diff
@@ -14,19 +14,24 @@
 
 namespace ConductorSharp.Engine.Builders
 {
+    public class DefinitionContext
+    {
+        public WorkflowOptions WorkflowOptions { get; set; }
+
+        public List<ITaskBuilder> TaskBuilders { get; set; } = new();
+    }
+
     public class WorkflowDefinitionBuilder<TWorkflow> where TWorkflow : ITypedWorkflow
     {
         private readonly Type _workflowType;
         private string _name;
         private JObject _inputs = new();
-        private WorkflowOptions _workflowOptions;
-
-        private List<ITaskBuilder> _taskBuilders = new();
+        public DefinitionContext Context { get; set; } = new();
```

Comment:
This object is initialized here. It does not make sense for someone to set this property outside this class. Remove `set`